zigpy/ota/__init__.py

```diff
@@ -6,42 +6,87 @@
 import attr
 
 from zigpy.config import CONF_OTA, CONF_OTA_DIR, CONF_OTA_IKEA, CONF_OTA_LEDVANCE
-from zigpy.ota.image import ImageKey, OTAImage
+from zigpy.ota.image import ImageKey, OTAImageHeader
 import zigpy.ota.provider
 from zigpy.ota.validators import check_invalid
+import zigpy.types as t
 from zigpy.typing import ControllerApplicationType
 import zigpy.util
 
 LOGGER = logging.getLogger(__name__)
 
-DELAY_EXPIRATION = datetime.timedelta(hours=2)
 TIMEDELTA_0 = datetime.timedelta()
+DELAY_EXPIRATION = datetime.timedelta(hours=2)
 
 
 @attr.s
-class CachedImage(OTAImage):
+class CachedImage:
+    MAXIMUM_DATA_SIZE = 40
     DEFAULT_EXPIRATION = datetime.timedelta(hours=18)
 
+    image = attr.ib(default=None)
     expires_on = attr.ib(default=None)
+    cached_data = attr.ib(default=None)
 
     @classmethod
-    def new(cls, img: OTAImage):
+    def new(cls, img):
```

Comment:
what's the type of `img` ? can we add it?

Reply:
Whoops. I think I removed it because there was a circular import at some point in time. It'd be `BaseOTAImage`.

Reply:
Yeah, Circular imports with typing are PITA. Feel free to add that type to `zigpy.typing` then